src/ppl/src/Slots.h

```diff
@@ -95,6 +106,22 @@ struct Section
   int used_slots;
   int num_slots;
   Edge edge;
+
+  int getMaxContiguousSlots(const std::vector<Slot>& slots)
+  {
+    int max_contioguous_slots = std::numeric_limits<int>::min();
+    for (int i = begin_slot; i <= end_slot; i++) {
+      int contiguous_slots = 0;
+      while (i <= end_slot && !slots[i].blocked && !slots[i].used) {
```

Comment:
I see the blocked||used idiom in several places.  It would be nice to make a method on Slot for that.